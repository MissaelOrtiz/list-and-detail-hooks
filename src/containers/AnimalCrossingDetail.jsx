import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { fetchVillager } from '../services/animalCrossingApi';

const AnimalCrossingDetail = () => {
  const { id } = useParams();
  const [loading, setLoading] = useState(true);
  const [villager, setVillager] = useState({});

  useEffect(() => {
    fetchVillager(id)
      .then((villager) => setVillager(villager))
      .finally(() => setLoading(false));
  }, [id]);

  if(loading) return <h1>Now Loading...</h1>;
  return (
    <div>
      <img src={villager.image} alt={villager.name} height={'400px'}/>
      <p>{villager.name}</p>
      <p>{villager.gender}</p>
      <p>{villager.quote}</p>
      <p>{villager.species}</p>
      <p>{villager.birthday}</p>
      <p>{villager.phrase}</p>
      <p>{villager.style}</p>
    </div>
  );
};

export default AnimalCrossingDetail;
