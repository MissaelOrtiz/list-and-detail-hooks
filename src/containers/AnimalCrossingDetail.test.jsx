/* eslint-disable max-len */
import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import AnimalCrossingDetail from './AnimalCrossingDetail';
import listData from '../fixtures/detailListData.json';

// const villagerId = '5f5fb4bbbfd05c2aed82e460';
const server = setupServer(
  rest.get('https://ac-vill.herokuapp.com/villagers/:id', (req, res, ctx) => {
    return res(ctx.json(listData));
  })
);

describe('AnimalCrossingDetail', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('displays info about a single villager', async () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/5f5fb4bbbfd05c2aed82e460']} keyLength={25}>
        <AnimalCrossingDetail />
      </MemoryRouter>
    );

    screen.getByText('Now Loading...');

    await screen.findByText('Admiral', { exact: false });
    expect(container).toMatchSnapshot();
  });
});
