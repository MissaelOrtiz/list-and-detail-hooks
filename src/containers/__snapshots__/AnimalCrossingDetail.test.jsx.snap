// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimalCrossingDetail displays info about a single villager 1`] = `
<div>
  <div
    aria-label="villager"
    role="div"
  >
    <img
      height="400px"
    />
    <p />
    <p />
    <p />
    <p />
    <p />
    <p />
    <p />
  </div>
</div>
`;
